<launch>
<!-- ============ Arguments ============ -->
  <!-- General -->
  <arg name="namespace" default="data"/> 
  <arg name="use_rio" default="false"/>
  <arg name="use_detectron" default="false"/>
  <arg name="visualize" default="true"/> 
  
  <!-- Flat Dataset args -->
  <arg name="base_path" default="/home/lukas/Documents/Datasets/flat_dataset/run2"/> 
  <arg name="play_rate" default="5"/>  <!-- Realtime factor -->
  <arg name="wait_time" default="0"/>  <!-- s -->
  <arg name="max_frames" default="1000"/> <!-- 90 -->
  
  <!-- Rio Dataset args -->
  <arg name="data_path" default="/home/lukas/Documents/Datasets/3RScan"/> 
  <arg name="scene_id" default="0"/> <!-- 0: 0-3, 1: 0-3, 2: 0-1 -->
  <arg name="scan_id" default="0"/>
  <arg name="rio_play_rate" default="20"/>  <!-- Hz -->
    
  <!-- Mapper -->
  <arg name="config" default="flat_groundtruth"/> <!-- flat_groundtruth, flat_detectron, rio_groundtruth, rio_detectron -->
  <arg name="shutdown_when_finished" default="false"/> 
  
  <!-- Save/Load maps -->
  <arg name="load_map" default = "false"/>
  <arg name="load_file" default = "/home/lukas/Documents/PanopticMapping/Data/maps/run1.panmap"/>
  <arg name="save_file" default = "/home/lukas/Documents/PanopticMapping/Data/maps/new_map"/>



<!-- ============ Run ============ -->
  <!-- Play data -->
  <arg name="wait" default = "true" if="$(arg load_map)"/>
  <include file="$(find panoptic_mapping_utils)/launch/play_flat_dataset.launch" pass_all_args="true" unless="$(arg use_rio)" />
  <include file="$(find panoptic_mapping_utils)/launch/play_rio_dataset.launch" pass_all_args="true" if="$(arg use_rio)" />  
  
  <!-- Mapper -->
  <node name="panoptic_mapper" pkg="panoptic_mapping_ros" type="panoptic_mapper_node" output="screen" ><!--required="true"-->
    <!-- Config -->
    <rosparam file="$(find panoptic_mapping_ros)/config/mapper/$(arg config).yaml"/>
    <param name="save_map_path_when_finished" value="$(arg save_file)" />  
    <param name="shutdown_when_finished" value="$(arg shutdown_when_finished)" /> 
     
    <!-- Input -->
    <remap from="color_image_in" to="$(arg namespace)/color_image"/>
    <remap from="depth_image_in" to="$(arg namespace)/depth_image"/>
    <remap from="segmentation_image_in" to="$(arg namespace)/segmentation_image"/>
    <remap from="labels_in" to="$(arg namespace)/segmentation_labels"/>
  </node>  
  
  <!-- Map loader -->
  <node name="map_loader" pkg="panoptic_mapping_utils" type="map_loader.py" if="$(arg load_map)" output="screen">
     <param name="path" value="$(arg load_file)" /> 
     <param name="srv_name" value="/panoptic_mapper/load_map" /> 
     <param name="delay" value="0.1" /> 
  </node>  
  
  <!-- RVIZ Visualization-->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find panoptic_mapping_ros)/config/rviz/devel.rviz" output="screen" if="$(arg visualize)"/>

</launch>
